core/http-auth/src/main/java/software/amazon/awssdk/http/auth/NoAuthAuthScheme.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * No-auth auth scheme ID.
+ */
+@SdkPublicApi
+public final class NoAuthAuthScheme {
```

Comment:
I think we make this implement `AuthScheme` like the other schemes. The `signer` and `identityProvider` methods can throw IllegalStateException because they are not expected to be called, or maybe return null.

Since `sdk-core` does not depend on `http-auth` and only `http-auth-spi`, maybe this class could live in `http-auth-spi`.

Reply:
For the `T extends Identity`, we could do `public final class NoAuthAuthScheme implements AuthScheme<Identity>`